### PR TITLE
Added rendering special case to integral tiles with sparse histograms.

### LINKED ISSUE #517: Documentation Results Issues
Refer to this build: https://3415-174014757-gh.circle-artifacts.com/0/rf-site/index.html

* In `unsupervised-learning.pymd` the final prediction raster is a single color:
image

* In `masking.pymd` the masks from `SCL` aren't rendered properly as tiles. However, application of mask looks correct:
image

* In `raster-join.pymd` need to confirm the MODIS columns is correct.... looks like multiple resolutions are being combined?
image

### PR DESCRIPTION
**Pull request by metasim**
Closes #517.
Depends on #516.

## `unsupervised-learning.ipymd`
image

## `masking.pymd`

image

* [x] Updated release-notes.md

### CODE REVIEW

**Comment by metasim** [comment]
Rendered docs:
https://3425-174014757-gh.circle-artifacts.com/0/rf-site/index.html

**Comment by metasim** [comment]
Looks like there's an outstanding issue:

image

**Comment by vpipkt** [comment]
@metasim which doc page is this?

> Looks like there's an outstanding issue:
> 
> image

**Comment by metasim** [comment]
@vpipkt 

> To consider, the clip_raster tiles contain either nodata or 1 in all cells

Yes, that seems to ring true. I think the bug is here:

https://github.com/locationtech/rasterframes/pull/518/files#diff-16f6c3a67d2e3cd105030cc6383a071077014afd88bbbe2493f2cab7b31d7c18R229

I was trying to deal with the case where the denominator would be zero. Thoughts on how we should handle this case?

File: core/src/test/scala/org/locationtech/rasterframes/functions/TileFunctionsSpec.scala [hunk 1]
```diff
@@ -516,10 +519,58 @@ class TileFunctionsSpec extends TestEnvironment with RasterMatchers {
...
 
       val pngData = expr.first()
-
+      // Transform it back into a raster.
       val image = ImageIO.read(new ByteArrayInputStream(pngData))
       image.getWidth should be(red.cols)
       image.getHeight should be(red.rows)
     }
+
+    it("should distribute classes across color ramp") {
```
Comment (vpipkt) [inline core/src/test/scala/org/locationtech/rasterframes/functions/TileFunctionsSpec.scala:528]: this is really cool. we add a few edge cases along this line:

 - tile is full of nodata
 - tile is nodata and 1 other value
 - tile is only 1 value

**Review by vpipkt — COMMENTED** [review]
just a thought about the "clipping" one ... nothing to helpful about how to fix

**Comment by vpipkt** [comment]
ya the `display(tile)` seems to result in dupe outputs of the tile in HTML. there are a few more to clean up

**Comment by metasim** [comment]
Much better:
image

**Review by vpipkt — APPROVED** [review]
